Shared/Samples/Analyze hotspots/AnalyzeHotspotsView.swift

```diff
@@ -0,0 +1,163 @@
+    )
+    /// The error thrown by geoprocessing job.
+    @State private var analysisError: Error?
+    /// A Boolean value indicating whether the geoprocessing job is running.
+    @State private var isAnalyzing: Bool = false
+    /// A Boolean value that indicates whether the analysis sheet is presented.
+    @State private var isAnalysisSheetPresented = false
+    /// The from date for the analysis.
+    @State private var fromDate = Self.minDate
+    /// The to date for the analysis.
+    @State private var toDate = Self.maxDate
+    
+    /// The date range for the analysis.
+    var dateRange: ClosedRange<Date> { fromDate...toDate }
+    
+    /// The minimum date for the analysis.
+    static var minDate: Date {
+        Calendar.current.date(from: DateComponents(year: 1998, month: 1, day: 1))!
+    }
+    /// The maximum date for the analysis.
+    static var maxDate: Date {
+        Calendar.current.date(from: DateComponents(year: 1998, month: 05, day: 31))!
+    }
+    
+    var body: some View {
+        MapViewReader { mapView in
+            MapView(map: map)
+                .task(id: dateRange) {
+                    map.removeAllOperationalLayers()
+                    // Creates and configures the job's parameters.
+                    let parameters = makeParameters(
+                        spatialReference: map.spatialReference,
+                        dateRange: dateRange
+                    )
+                    // Creates and runs the job.
+                    let geoprocessingJob = geoprocessingTask.makeJob(parameters: parameters)
+                    isAnalyzing = true
+                    geoprocessingJob.start()
+                    do {
+                        let result = try await withTaskCancellationHandler {
+                            try await geoprocessingJob.output
+                        } onCancel: {
+                            Task.detached {
+                                await geoprocessingJob.cancel()
+                            }
+                        }
+                        guard let mapImageLayer = result.mapImageLayer else {
+                            return
+                        }
+                        try await mapImageLayer.load()
+                        map.addOperationalLayer(mapImageLayer)
+                        isAnalyzing = false
+                        if let extent = mapImageLayer.fullExtent {
+                            await mapView.setViewpointGeometry(extent)
+                        }
+                    } catch {
+                        analysisError = error
+                    }
+                }
+                .toolbar {
+                    if isAnalyzing {
+                        ToolbarItem(placement: .topBarTrailing) {
+                            ProgressView()
+                                .progressViewStyle(.circular)
+                        }
+                    }
```

Comment:
```suggestion
                }
                .overlay(alignment: .center) {
                    if isAnalyzing {
                        VStack {
                            Text("Populating")
                            ProgressView()
                                .progressViewStyle(.circular)
                        }
                        .padding()
                        .background(.ultraThickMaterial)
                        .clipShape(.rect(cornerRadius: 10))
                        .shadow(radius: 50)
                    }
                }
                .toolbar {
```

The progress view could be centered over the view rather than placed in the toolbar to align more with other sample designs. "Populating" is used here as an example but can be changed or omitted.


<img src="https://github.com/user-attachments/assets/a6c37633-8935-49d8-a89f-f0f118f8cf91" width="300">

Reply:
Thanks for clarifying.